### PR TITLE
docs: fix 404 links to sytle guide

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
-  fix 404 links to sytle guide

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPdocs
- [] Unit testing, with >80% coverage
- [] User guide updated
- [] Conforms to style guide

### CODE REVIEW

File: contributing/guidelines.rst [hunk 1]
```diff
@@ -11,7 +11,7 @@ PHP Style
 =========
 
 All code must conform to our `Style Guide
-<./styleguide.rst>`_, which is
+<./styleguide.md>`_, which is
 essentially the `Allman indent style
```
Comment (paulbalandan) [inline contributing/guidelines.rst:15]: This should no longer be Allman, I think. Perhaps, the K&R style now applies.

Comment (MGatner) [reply contributing/guidelines.rst:15]: Can we just say "PSR-12" instead?

Comment (ytetsuro) [reply contributing/guidelines.rst:15]: Are you going to use this PR to correct items that are not related to the purpose of the PR?

Comment (kenjis) [reply contributing/guidelines.rst:15]: @paulbalandan @MGatner I will update the entire Contribution Guidelines in another PR.

Comment (kenjis) [reply contributing/guidelines.rst:15]: We have two Guidelines:
- https://github.com/codeigniter4/CodeIgniter4/blob/develop/CONTRIBUTING.md
- https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/guidelines.rst

It seems it is better to merge into one (CONTRIBUTING.md).

Comment (kenjis) [reply contributing/guidelines.rst:15]: No, instead of merging them into one, it might be better to change them to eliminate duplication of content.

**Review by paulbalandan — APPROVED** [review]

**Comment by kenjis** [comment]
I will update the entire Contribution Guideline files to eliminate duplication of content in another PR.
Merge this PR.

**Pull request merged**